content/concepts/introduction/protocols/multistream.md

```diff
@@ -1,7 +1,66 @@
 ---
 title: "Multistream Select"
-description: ""
+description: "Multistream Select is used to negotiate and upgrade the protocol of a connection between two peers in libp2p."
 weight: 26
 ---
 
-Coming soon!
+## Overview
+
+[Multistream Select](https://github.com/multiformats/multistream-select) is
+a protocol for friendly protocol multiplexing that allows for the negotiation
+of a [multicodec](https://github.com/multiformats/multicodec) between two entities.
+
+> Multicodecs are codecs that can handle multiple different encoding formats.
+> They are used to decode data and are self-describing, meaning they can identify
+> the data format they are decoding. This capability allows nodes to support multiple
+> protocols and efficiently switch between them depending on the task.
+
+Multistream is a general concept and refers to the ability
+to [multiplex multiple streams](../../multiplex/overview.md) of data over a single
+connection. This allows for multiple independent channels of communication to be
+established over a single connection rather than needing to open a separate
+connection for each channel.
```

Comment:
That's multiplexing, not multistream. I don't think we need to explain multiplexing here though.